### PR TITLE
msp: provision read-only 'operator access' user for Cloud SQL

### PR DESCRIPTION
**Pull request by bobheadxi**
This change adds an "operator access" service account that I plan to expand with various roles and stuff required to connect to MSP resources on a read-only basis. Currently, I'm focusing on the Cloud SQL use case as part of https://github.com/sourcegraph/managed-services/issues/207 for a `mi2`-like UX for connecting to databases, so I've also added SQL users for the operator access SA. For escalated access, operators will impersonate the existing Cloud Run workload SA instead.

From here we can impersonate the SA to connect to the instance using the connection name that I've added to the Cloudrun stack outputs in this PR - https://github.com/sourcegraph/sourcegraph/pull/59106

## Test plan

- https://github.com/sourcegraph/sourcegraph/pull/59106
- https://github.com/sourcegraph/managed-services/pull/220

### CODE REVIEW

**Comment by bobheadxi** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #59105** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/59105?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #59106** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/59106?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/59105?utm_source=stack-comment).

**Review by unknwon — APPROVED** [review]
Assuming it works

File: dev/managedservicesplatform/internal/resource/postgresqlroles/postgresqlroles.go [hunk 1]
```diff
@@ -52,6 +52,13 @@ func New(scope constructs.Construct, id resourceid.ID, config Config) (*Output,
 		WithAdminOption: jsii.Bool(true),
 	})
 
+	_ = grantrole.NewGrantRole(scope, id.TerraformID("operator_access_service_account_role_cloudsqlsuperuser"), &grantrole.GrantRoleConfig{
+		Provider:        pgProvider,
+		Role:            config.OperatorAccessUser.Name(),
+		GrantRole:       jsii.String("cloudsqlsuperuser"),
+		WithAdminOption: jsii.Bool(true),
+	})
+
```
Comment (michaellzc) [inline dev/managedservicesplatform/internal/resource/postgresqlroles/postgresqlroles.go:61]: have you tested db access using the operator access user? afaik, granting the `cloudsqlsuperuser` won't work. last time I checked, it needs to be granted read access to individual database manually, see controller code.

Comment (bobheadxi) [reply dev/managedservicesplatform/internal/resource/postgresqlroles/postgresqlroles.go:61]: Huh - I don't grant the workload user anything

Comment (michaellzc) [reply dev/managedservicesplatform/internal/resource/postgresqlroles/postgresqlroles.go:61]: it has something to do with postgres db permission. only the creator of the table `create table` has access by default, or the user that creator the table creator (the admin user).

it's a bit fuzzy haha, I just got something working without looking into it too hard

Comment (bobheadxi) [reply dev/managedservicesplatform/internal/resource/postgresqlroles/postgresqlroles.go:61]: ah, you were right - fixing

Comment (bobheadxi) [reply dev/managedservicesplatform/internal/resource/postgresqlroles/postgresqlroles.go]: Updated, but currently running into:

```
│ Error: Error granting role pg_database_owner to msp-admin: pq: role "pg_database_owner" cannot have explicit members
```

Comment (bobheadxi) [reply dev/managedservicesplatform/internal/resource/postgresqlroles/postgresqlroles.go]: Relevant changes should all be in https://github.com/sourcegraph/sourcegraph/pull/59105/files#diff-e969547f3a1bd376cce3c4f2ba57e0958037ff6ce39edceb89e3869840b66df6

**Comment by bobheadxi** [comment]
### Merge activity

* **Dec 20, 5:24 PM**: @@bobheadxi merged this pull request with [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/59105).

**Pull request merged**